apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -619,6 +623,73 @@ flush(Data0) ->
             end
     end.
 
+stop_before_mixed_stop_after_render(
```

Comment:
maybe this is more readable:
```
stop_before_mixed_stop_after_render(?QUERY(_, _, _, _, TraceCtx), State) ->
    do_stop_before_mixed_stop_after_render(TraceCtx, State).

do_stop_before_mixed_stop_after_render(TraceCtx, initial_state) ->
    do_stop_before_mixed_stop_after_render(TraceCtx, get_stop_flag(TraceCtx));
do_stop_before_mixed_stop_after_render(TraceCtx, StopFlag) ->
    case get_stop_flag(TraceCtx) =:= StopFlag  of
        true ->
            StopFlag;
        false ->
            true
    end.

get_stop_flag(#{stop_action_after_render := true}) -> stop_action_after_render;
get_stop_flag(_) -> no_stop_action_after_render.
```

Reply:
Thanks, I like your suggestion. I think it needs some small fixes and I will change the name to stop_batching.

Reply:
Refactored thanks to your suggestion: [b16b9d8](https://github.com/emqx/emqx/pull/12827/commits/b16b9d8fcc3a4a3ca36f0c30f341c88995d8470c). Thanks, I think it is much more readable now.